src/ddsrt/include/dds/ddsrt/rusage.h

```diff
@@ -14,20 +14,7 @@
 
 #include <stddef.h>
 
-#if DDSRT_WITH_FREERTOS
```

Comment:
The entire (and *very* ugly) `check_macro` CMake stuff is used solely for determining whether `rusage` is available on FreeRTOS. I think that's unwise and this bit of pre-processing is the better option for this case.

It is trivial to turn FreeRTOS' rusage.c into an empty file if it isn't supported for the particular FreeRTOS configuration which avoids having to know in CMake whether or not `rusage` is supported.

Reply:
Well, it's not so much about whether or not we can compile `rusage.c` in case of FreeRTOS, it's more about other parts of cyclone knowing whether or not rusage is available. If I were to determine it in e.g. `rusage.h` or `config.h`, that'd mean I need to include `FreeRTOS.h` to determine the same thing. Which, imho is not much different from leaking `windows.h`(?)

I can see why you'd describe `check_macro` as ugly (or even *very* ugly), but it's not all that different from other compiler checks carried out during the configuration process right(?) At the core it provides a concise way to check for something that depends on preprocessor checks and might be useful once we decide to not include system headers in cyclone's headers(?)

I'm happy to reinstate the previous way of checking things, just thought the aforementioned might not have been taken into consideration. Let me know your preference and I'll make the necessary changes.